funsor/tensor.py

```diff
@@ -70,8 +72,11 @@ class TensorMeta(FunsorMeta):
     Wrapper to fill in default args and convert between OrderedDict and tuple.
     """
     def __call__(cls, data, inputs=None, dtype="real"):
+        # FIXME: how to make the following assertation pass?
+        #   type(Tensor(np.ones(3))) is type(Tensor(onp.ones(3)))
```

Comment:
@fritzo Do you have a solution for this issue? This is not important I guess...

Reply:
@fehiepsi why do you need this to be true when `type(np.ones(3)) is type(onp.ones(3))` is false?

Reply:
@eb8680 This is just to cleanup the assert_close test. :D

Reply:
To clarify, types in Funsor work a little differently than in regular Python to support pattern matching - `type(Tensor(np.ones(3)))` [evaluates](https://github.com/pyro-ppl/funsor/blob/master/funsor/terms.py#L222) via [`reflect`](https://github.com/pyro-ppl/funsor/blob/master/funsor/terms.py#L76) `Tensor[type(np.ones(3))]`, not `Tensor`.  If you wanted to change this behavior for `Tensor`, you'd need to write a custom `TensorMeta.__getitem__` that made your assertion true by construction:
```py
def __getitem___(cls, arg_types):
    arg_types = (arg_types,) if not isinstance(arg_types, tuple) else arg_types
    arg_types = (GENERIC_TENSOR_TYPE,) + arg_types[1:]  # e.g. onp.generic
    return super().__getitem__(cls, arg_types)
```
However, it's not clear to me why the above should be true - I don't think `Tensor` should be converting data between different backends or asserting that they're the same.

Reply:
Understood. I just remove those comments and the cast from onp.generic to onp.ndarray too. Thanks for explaining!